bus-mapping/src/circuit_input_builder.rs

```diff
@@ -255,8 +318,22 @@ impl<'a, C: CircuitsParams> CircuitInputBuilder<C> {
                 &geth_trace.struct_logs[index..],
             )?;
             tx.steps_mut().extend(exec_steps);
-        }
 
+            // Generate EndChunk and proceed to the next if it's not the last chunk
+            if is_chunked && self.chunk_ctx.rwc.0 >= self.circuits_params.max_rws() {
```

Comment:
Update:
I implemented a design to introduce new padding https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1718 to address this issue.

With new execution::padding, the boundary condition to trigger chunk become
- current_rws + next_step_rws + end_chunk_rws >= `max_rws` => need to chunk now and extend next_step to new chunk 
- estimated current evm_row + max_row_of_next_step >= `max_evm_row` => Maybe we do not need to handle this case because before once reach `max_evm_row` we already exceed max_gas_per_block beforehand

Reply:
Thanks for the implementation! If you want to add look ahead + chunk padding like this its fine :) but would you consider my original design? As I explained in last comments, the `max_rws` is actually addressed with a reset mechanism, not look ahead. When I worked on this part I felt that resetting is easier, less proned to bug. Lmk what you think?

Reply:
> Here we need to handle 2 corner cases
> 
> 1. evm steps reach `max_evm_rows` earlier than rw_table reach `max_rws`. we need to take max_height of step as a parameter into consideration.
> 2. rw_table reach `max_rws` earlier than evm steps reach `max_evm_rows`. we also need to reserve extra rws for end chunk
> 
honestly idk how to handle this, the evm_rows isn't calculated till the EvmCircuit is config, right? i don't see much possibility of moving that logic over from [here](https://github.com/privacy-scaling-explorations/zkevm-circuits/blob/853624b53551a75b7db627dcc75b353a345ad52a/zkevm-circuits/src/evm_circuit.rs#L258) cuz ultimately it relies on [EXECUTION_STATE_HEIGHT_MAP](https://github.com/privacy-scaling-explorations/zkevm-circuits/blob/5810e7337a63d3aa49db208029082ae2feb11723/zkevm-circuits/src/evm_circuit/param.rs#L170) which does not exist in bus-mapping

Reply:
Yeah case 1 based on current design we can only set as parameter and have a assertion to check max height match with it in EVM Circuit.

But I think maybe we can ignore case 1 and just handle case 2, and assuming before reaching max_evm_rows the max gas per block will be explode first. Based on our previously benchmark, `max_rws` is easily to be abused than `max_evm_row`.